### PR TITLE
New Integration - SMIME Messaging

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/17269

## Description
A few sentences describing the overall goals of the pull request's commits.

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](link)

### CODE REVIEW

**Comment by Shellyber** [comment]
@kirbles19 Please take a look at the YML file

File: Integrations/SMIME_Messaging/SMIME_Messaging.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
+category: Authentication
+commonfields:
+  id: SMIME_Messaging
```
Comment (Itay4) [inline Integrations/SMIME_Messaging/SMIME_Messaging.yml]:
```suggestion
  id: SMIME Messaging
```

File: Integrations/SMIME_Messaging/SMIME_Messaging.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  name: public_key
+  required: true
+  type: 12
+- display: Private Key
+  name: private_key
+  required: true
+  type: 14
+description: S/MIME - Secure Multipurpose Internet Mail Extensions- provides a consistent
+  way to send and receive secure MIME data.
+display: SMIME_Messaging
```
Comment (Itay4) [inline Integrations/SMIME_Messaging/SMIME_Messaging.yml]: SMIME Messaging

File: Integrations/SMIME_Messaging/SMIME_Messaging.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+name: SMIME_Messaging
```
Comment (Itay4) [inline Integrations/SMIME_Messaging/SMIME_Messaging.yml]: SMIME Messaging

File: Integrations/SMIME_Messaging/SMIME_Messaging.yml [hunk 4]
```diff
@@ -0,0 +1,92 @@
+category: Authentication
```
Comment (Itay4) [inline Integrations/SMIME_Messaging/SMIME_Messaging.yml]:
```suggestion
category: Utilities
```

File: Integrations/SMIME_Messaging/SMIME_Messaging.yml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+configuration:
+- display: Public Key
+  name: public_key
+  required: true
+  type: 12
+- display: Private Key
+  name: private_key
+  required: true
+  type: 14
+description: S/MIME - Secure Multipurpose Internet Mail Extensions- provides a consistent
```
Comment (Itay4) [inline Integrations/SMIME_Messaging/SMIME_Messaging.yml]:
```suggestion
description: S/MIME (Secure Multipurpose Internet Mail Extensions) provides a consistent
```

File: Integrations/SMIME_Messaging/SMIME_Messaging.py [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+        'SMIME': {
+            'Message': out,
+        }
+    }
+    human_readable = f'The encrypted message is: \n{out}'
+
+    return human_readable, entry_context
+
+
+def test_module():
```
Comment (Itay4) [inline Integrations/SMIME_Messaging/SMIME_Messaging.py]: can we do something in the test? maybe sign?

File: Integrations/SMIME_Messaging/SMIME_Messaging.py [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+    # Output p7 in mail-friendly format.
+    out = BIO.MemoryBuffer()
+    client.smime.write(out, p7, buf)
+
+    signed = out.read().decode('utf-8')
+    signed_message = signed.split('\n\n')
+    headers = signed_message[0].replace(': ', '=').replace('\n', ',')
+    body = signed_message[2:]
+    context = {
+        'SMIME': {
```
Comment (Itay4) [inline Integrations/SMIME_Messaging/SMIME_Messaging.py]:
```suggestion
        'SMIME.Signed': {
```

File: Integrations/SMIME_Messaging/SMIME_Messaging.py [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+    out = BIO.MemoryBuffer()
+
+    client.smime.write(out, p7)
+    encrypted_message = out.read().decode('utf-8')
+    message = encrypted_message.split('\n\n')
+    headers = message[0]
+    new_headers = headers.replace(': ', '=').replace('\n', ',')
+
+    entry_context = {
+        'SMIME': {
```
Comment (Itay4) [inline Integrations/SMIME_Messaging/SMIME_Messaging.py]:
```suggestion
        'SMIME.Encrypted': {
```

File: Integrations/SMIME_Messaging/SMIME_Messaging.py [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+    client.smime.load_key(client.private_key_file, client.public_key_file)
+
+    # Load the encrypted data.
+    p7, data = SMIME.smime_load_pkcs7(encrypt_message['path'])
+
+    # Decrypt p7.
+    out = client.smime.decrypt(p7).decode('utf-8')
+    entry_context = {
+        'SMIME': {
+            'Message': out,
```
Comment (Itay4) [inline Integrations/SMIME_Messaging/SMIME_Messaging.py]:
```suggestion
            'Message': out
```

File: Integrations/SMIME_Messaging/SMIME_Messaging.py [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+    # Load private key and cert.
+    client.smime.load_key(client.private_key_file, client.public_key_file)
+
+    # Load the encrypted data.
+    p7, data = SMIME.smime_load_pkcs7(encrypt_message['path'])
+
+    # Decrypt p7.
+    out = client.smime.decrypt(p7).decode('utf-8')
+    entry_context = {
+        'SMIME': {
```
Comment (Itay4) [inline Integrations/SMIME_Messaging/SMIME_Messaging.py]:
```suggestion
        'SMIME.Decrypted': {
```

File: Integrations/SMIME_Messaging/SMIME_Messaging.py [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+            'Message': out,
+        }
+    }
+    human_readable = f'The encrypted message is: \n{out}'
```
Comment (Itay4) [inline Integrations/SMIME_Messaging/SMIME_Messaging.py]:
```suggestion
    human_readable = f'The decrypted message is: \n{out}'
```

File: Integrations/SMIME_Messaging/SMIME_Messaging_test.py [hunk 12]
```diff
@@ -0,0 +1,42 @@
+from SMIME_Messaging import Client, sign_email, encrypt_email_body, verify, decrypt_email_body
+import demistomock as demisto
+
+
+with open('./test_data/signer_key.pem') as f:
+    private_key = f.read()
+with open('./test_data/signer.pem') as file:
```
Comment (Itay4) [inline Integrations/SMIME_Messaging/SMIME_Messaging_test.py]: dont use reserved word like `file`

**Comment by kirbles19** [comment]
@Shellyber done.

File: Integrations/SMIME_Messaging/SMIME_Messaging.py (hunk 6 above)
Comment (Shellyber) [reply Integrations/SMIME_Messaging/SMIME_Messaging.py]: I called in the test_module the encrypt and decrypt so we can check that the keys are correct. 
Also added a call to this function in the unittest

File: Integrations/SMIME_Messaging/SMIME_Messaging.yml [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+display: SMIME Messaging
+name: SMIME Messaging
+script:
+  commands:
+  - arguments:
+    - default: false
+      description: The message body to send.
+      isArray: false
+      name: message_body
+      required: false
```
Comment (Itay4) [inline Integrations/SMIME_Messaging/SMIME_Messaging.yml]:
```suggestion
      required: true
```

File: Integrations/SMIME_Messaging/SMIME_Messaging.py [hunk 14]
```diff
@@ -0,0 +1,206 @@
...
+    client.smime.set_x509_stack(sk)
+
+    st = X509.X509_Store()
+    st.load_info(client.public_key_file)
+    client.smime.set_x509_store(st)
+
+    p7, data = SMIME.smime_load_pkcs7(signed_message['path'])
+    v = client.smime.verify(p7, data, flags=SMIME.PKCS7_NOVERIFY)
+
+    human_readable = f'The signature verified\n\n{v}\n{data}'
```
Comment (Itay4) [inline Integrations/SMIME_Messaging/SMIME_Messaging.py]:
```suggestion
    human_readable = f'The signature verified\n\n{v}'
```

File: Integrations/SMIME_Messaging/SMIME_Messaging.yml [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+      type: String
+    - contextPath: SMIME.Signed.Headers
+      description: The S/MIME signing headers.
+      type: String
+  - arguments:
+    - default: false
+      description: The message body to encrypt.
+      isArray: false
+      name: message
+      required: false
```
Comment (Itay4) [inline Integrations/SMIME_Messaging/SMIME_Messaging.yml]:
```suggestion
      required: true
```

**Pull request merged**